juju/juju
# Work on adding support for AWS instance profiles

Adds support for attaching Instance Profiles.

This change introduces the ability to attach already made and new Instance Profiles to Juju controller machines using Bootstrap constraints.

With this change a user can specify and already existing instance profile with --bootstrap-constraints="instance-role=test" or  --bootstrap-constraints="instance-role=auto" for the Juju bootstrap process to make a brand new instance profile in the form of juju-controller-<controller-name>.

This PR does not make the controller use these new credentials it just introduces the attachment and creation process.

## Checklist


## QA steps

Run Juju with bootstrap to aws with either:

- --bootstrap-constraints="instance-role=test" (where test is the name of an already create instance profile or
- --bootstrap-constraints="instance-role=auto" (to have Juju make a new Instance Profile)

You can learn about creating your own instance profiles here: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html

You can verify the attachment of the instance profile to the controller using: https://aws.amazon.com/premiumsupport/knowledge-center/attach-replace-ec2-instance-profile/

## Documentation changes

Not just yet
